packages/flutter/lib/src/widgets/animated_raster.dart

```diff
@@ -0,0 +1,226 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'media_query.dart';
+
+/// A delegate that controls what is painted during an animation in the [AnimatedRaster]
+/// widget.
+@immutable
+abstract class AnimatedRasterDelegate {
```

Comment:
This API probably needs another pass. paint and paintRaster should be more similar.